lsm/lsm.cpp

```diff
@@ -31,38 +128,42 @@ std::optional<int32_t> LSM::get(int32_t k) {
       return std::nullopt;
     }
   } else {
-    for (auto i = files.rbegin(); i != files.rend(); ++i ) {
-      File file = *i;
-      if (file.bloomFilter.contains_key(k) && k >= file.min && k <= file.max) {
-        std::map<int32_t, maybe_value> file_map = read_from_file(file.filename);
-        auto it_m = file_map.find(k);
-        if (it_m != m.end()) {
-          if(!it_m->second.is_deleted) {
-            return it_m->second.v;
-          } else {
-            return std::nullopt;
+    for(unsigned i = levels.size() - 1; levels.size() > i; --i) {
+      Level& level = get_level(i);
```

Comment:
Let's rewrite this to use a reverse iterator through the level. We don't need `get_level` because we're only looking at the currently present levels:
```cpp
    for (auto it = levels.rbegin(); it != levels.rend(); ++it) {
      auto& level = *it;
      for (auto j = level.files.rbegin(); j != level.files.rend(); ++j) {
        auto& file = *j;
```